### ISSUE TITLE
lab2 - Review  by Matteo Pietro Pillitteri

### ISSUE THREAD

**Issue by Matteo-Pietro-Pillitteri** [Zafonte/computational-intelligence#2]
Hi francesca, i have some advice for you. I hope that this review can help you.
- I think that you could write in your README some extra notes in order to clarify:
  - in this kind of problem, what is an “individual”?
  - some motivation about the hyperparameters that you chose (e.g:  μ , λ ,sigma)
  - the choice to have a number of eras that is 100// λ

- In order to have “more learning” you can use more iteration in order to produce more generations. If the hyperparameters that you chose are well setted, and you algorithm does not reach an high accuracy, you can try to run it on more generation.

- From your README I think that it's interesting the way in which you used sigma for balancing between exploration and exploration.
(e.g weight = strategy4.get_weight() + np.random.normal(loc=0, scale=sigma+1, size=(strategy4.__len__())) )
I think that you can try to scale sigma in a different way:
  - At the beginning you can set sigma with an higher value, for example 0.01
  - Reduce it linearly over the eras. I do something similar in my code, so for each step in range(..) you can try to do something like:   σ = max(final_σ, initial_σ - (initial_σ - final_σ) * (step / total_num_epochs))

- It can be helpful to have  some print or some graph for see how the weights are changing during the esecution.

Some other comments:
- I really appreciate the pseudo code that you included in your README, it is helpful to understand your code. 
- I think that your python is not bad and that you have done a great job!

**Comment by Zafonte** [comment]
Thank you for your review and your advices!
